stack/stack.go

```diff
@@ -1,39 +1,91 @@
 package main
 
+//Imported Libraries
 import (
-	"errors"
 	"fmt"
 )
 
-// Stack implements stack DS
-type Stack struct {
-	list []int
+//Constants
+const x  = 10
+
+//Implementation of the stack structure
+type stack struct {
+	items [x] string
+}
+//pop from the stack
+func (a *stack) push (value string){
+	var found bool = false
+	for index,valuea := range a.items{
+		if valuea == "" {
+			a.items[index]=value
+			found=true
+			break
+		}
+	}
+	if found == false{
+		fmt.Printf("Stack is full, Try on PUSH something outside\n")
+	}
 }
 
-// Push pushes to stack
-func (s *Stack) Push(data int) error {
-	s.list = append(s.list, data)
-	return nil
+//pop form the stack
+func (a *stack) pop()  {
+	var empty bool = true
+	for index := range a.items{
+		if a.items[x-index-1]!="" {
+			a.items[x-index-1]=""
+			empty= false
+			break
+		}
+	}
+	if empty==true{
+		fmt.Printf("Stack is empty! Cannot pop! Try on push in something\n")
+	}
 }
 
-// Pop pops from stack
-func (s *Stack) Pop() (int, error) {
-	if len(s.list) == 0 {
-		return -1, errors.New("pop from empty stack")
+//print the stack
+func (a *stack) printstack()  {
```

Comment:
Formatting can be improved.

*Origin: GofmtBear, Section: `gofmt`.*

The issue can be fixed by applying the following patch:

```diff
--- a/tmp/tmpgx2e53q9/stack/stack.go
+++ b/tmp/tmpgx2e53q9/stack/stack.go
@@ -43,14 +43,14 @@
 }
 
 //print the stack
-func (a *stack) printstack()  {
+func (a *stack) printstack() {
 	fmt.Printf("(")
-	for index := range a.items{
+	for index := range a.items {
 		value := "None"
-		if a.items[index]!=""{
-			value=a.items[index]
+		if a.items[index] != "" {
+			value = a.items[index]
 		}
-		fmt.Printf("%s,",value)
+		fmt.Printf("%s,", value)
 	}
 	fmt.Println(")")
 }
```